tensorflow/core/grappler/optimizers/meta_optimizer.cc

```diff
@@ -147,21 +147,21 @@ Status MetaOptimizer::Optimize(Cluster* cluster, const GrapplerItem& item,
         "pruning",    "function",      "constfold",  "layout",
         "memory",     "autoparallel",  "arithmetic", "loop",
         "dependency", "debug_stripper"};
-    std::vector<string> custom_optimizer_names;
-    for (const auto& optimizer_name : cfg_.optimizers()) {
-      if (available_optimizers.find(optimizer_name) !=
+    std::vector<tensorflow::RewriterConfig_CustomGraphOptimizer> custom_optimizer_list;
+    for (const auto& optimizer : cfg_.custom_optimizers()) {
+      if (available_optimizers.find(optimizer.name()) !=
           available_optimizers.end()) {
-        optimizers.push_back(NewOptimizer(optimizer_name));
+        optimizers.push_back(NewOptimizer(optimizer.name()));
       } else {
-        custom_optimizer_names.push_back(optimizer_name);
+        custom_optimizer_list.push_back(optimizer);
       }
     }
     // Now run the custom optimizers.
-    for (const auto& optimizer_name : custom_optimizer_names) {
+    for (const auto& optimizer : custom_optimizer_list) {
```

Comment:
I think it would be best to simply add an extra loop at the end to go over the optimizers listed in 'cfg_.custom_optimizers()' and add them to the mix. This will ensure that they're run last, and therefore that you'll have a constant folded and layout optimized model to start with.